x-pack/plugins/lens/public/visualizations/gauge/toolbar_component/gauge_toolbar.test.tsx

```diff
@@ -5,136 +5,114 @@
 import { GaugeToolbar } from '.';
 import type { GaugeVisualizationState } from '../constants';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
-jest.mock('lodash', () => {
-  const original = jest.requireActual('lodash');
-
-  return {
-    ...original,
-    debounce: (fn: unknown) => fn,
+describe('gauge toolbar', () => {
+  const defaultProps: VisualizationToolbarProps<GaugeVisualizationState> = {
+    setState: jest.fn(),
+    frame: {} as FramePublicAPI,
+    state: {
+      layerId: 'layerId',
+      layerType: 'data',
+      metricAccessor: 'metric-accessor',
+      minAccessor: '',
+      maxAccessor: '',
+      goalAccessor: '',
+      shape: 'verticalBullet',
+      colorMode: 'none',
+      ticksPosition: 'auto',
+      labelMajorMode: 'auto',
+    },
   };
-});
-
-class Harness {
-  wrapper: ReactWrapper;
-
-  constructor(wrapper: ReactWrapper) {
-    this.wrapper = wrapper;
-  }
-
-  togglePopover() {
-    this.wrapper.find('button[data-test-subj="lnsVisualOptionsButton"]').simulate('click');
-  }
-
-  public get titleLabel() {
-    return this.wrapper.find('EuiFieldText[data-test-subj="lnsToolbarGaugeLabelMajor"]');
-  }
-  public get titleSelect() {
-    return this.wrapper.find('EuiSelect[data-test-subj="lnsToolbarGaugeLabelMajor-select"]');
-  }
-
-  modifyTitle(e: FormEvent) {
-    act(() => {
-      this.titleLabel.prop('onChange')!(e);
-    });
-  }
 
-  public get subtitleSelect() {
-    return this.wrapper.find('EuiSelect[data-test-subj="lnsToolbarGaugeLabelMinor-select"]');
-  }
-
-  public get subtitleLabel() {
-    return this.wrapper.find('EuiFieldText[data-test-subj="lnsToolbarGaugeLabelMinor"]');
-  }
-
-  modifySubtitle(e: FormEvent) {
-    act(() => {
-      this.subtitleLabel.prop('onChange')!(e);
-    });
-  }
-}
+  beforeEach(() => {
+    (defaultProps.setState as jest.Mock).mockClear();
+    jest.useFakeTimers();
+  });
 
-describe('gauge toolbar', () => {
-  let harness: Harness;
-  let defaultProps: VisualizationToolbarProps<GaugeVisualizationState>;
+  afterEach(() => {
+    jest.useRealTimers();
+  });
 
-  beforeEach(() => {
-    defaultProps = {
-      setState: jest.fn(),
-      frame: {} as FramePublicAPI,
-      state: {
-        layerId: 'layerId',
-        layerType: 'data',
-        metricAccessor: 'metric-accessor',
-        minAccessor: '',
-        maxAccessor: '',
-        goalAccessor: '',
-        shape: 'verticalBullet',
-        colorMode: 'none',
-        ticksPosition: 'auto',
-        labelMajorMode: 'auto',
-      },
+  const renderAxisTicksSettingsAndOpen = (
+    propsOverrides?: Partial<VisualizationToolbarProps<GaugeVisualizationState>>
+  ) => {
+    const rtlRender = render(<GaugeToolbar {...defaultProps} {...propsOverrides} />);
+    const openPopover = () => userEvent.click(screen.getByRole('button', { name: 'Appearance' }));
+    openPopover();
+    return {
+      ...rtlRender,
```

Comment:
why shallow copy?

Reply:
no particular reason, it was to prepare to return other custom methods if we needed them in the future.  But I can remove it and add when needed.